### PR TITLE
Consistently map licenses, copyright statements, and use statements for items and collections

### PR DESCRIPTION
**Pull request by mjgiarlo**
Connects to sul-dlss/argo#2404
Fixes sul-dlss/argo#2612

## Why was this change made?

Data loss is bad.

## How was this change tested?

CI. Should prolly be tested in QA by @andrewjbtw before we go forward.

## Which documentation and/or configurations were updated?

None

### CODE REVIEW

File: app/services/cocina/from_fedora/access.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ module Cocina
   module FromFedora
     # builds the Access subschema for Collections
     class Access
+      NONE_LICENSE_URI = 'http://cocina.sul.stanford.edu/licenses/none'
```
Comment (mjgiarlo) [inline app/services/cocina/from_fedora/access.rb:7]: I made this up, obviously.

File: app/services/cocina/from_fedora/access.rb [hunk 2]
```diff
@@ -20,7 +22,9 @@ def props
           access: access_rights,
           download: download? ? access_rights : 'none',
           readLocation: location,
-          license: license_uri
+          license: license_uri,
+          copyright: copyright,
+          useAndReproductionStatement: use_statement
```
Comment (mjgiarlo) [inline app/services/cocina/from_fedora/access.rb:27]: These previously were not mapped.

File: app/services/cocina/from_fedora/dro_access.rb [hunk 3]
```diff
@@ -16,8 +16,9 @@ def initialize(rights_metadata_ds, embargo:)
       def props
         super.tap do |access|
           access[:embargo] = embargo unless embargo.empty?
-          access[:useAndReproductionStatement] = rights_metadata_ds.use_statement.first if rights_metadata_ds.use_statement.first.present?
-          access[:copyright] = rights_metadata_ds.copyright.first if rights_metadata_ds.copyright.first.present?
+          access[:useAndReproductionStatement] = use_statement if use_statement.present?
+          access[:copyright] = copyright if copyright.present?
+          access[:license] = license_uri if license_uri.present?
```
Comment (mjgiarlo) [inline app/services/cocina/from_fedora/dro_access.rb:21]: This was not previously mapped.

File: app/services/cocina/to_fedora/access.rb [hunk 4]
```diff
@@ -23,12 +23,42 @@ def apply
...
+          item.rightsMetadata.creative_commons_human = Dor::CreativeCommonsLicenseService.property(license_code).label
+        elsif Dor::OpenDataLicenseService.key?(license_code)
+          item.rightsMetadata.open_data_commons = license_code
+          item.rightsMetadata.open_data_commons.uri = Dor::OpenDataLicenseService.property(license_code).uri
+          item.rightsMetadata.open_data_commons_human = Dor::OpenDataLicenseService.property(license_code).label
+        elsif license_code == 'none'
+          item.rightsMetadata.creative_commons = license_code
+          item.rightsMetadata.creative_commons_human = 'no Creative Commons (CC) license'
+        else
+          raise ArgumentError, "'#{license_code}' is not a valid license code"
```
Comment (mjgiarlo) [inline app/services/cocina/to_fedora/access.rb:49]: This logic is essentially ported from dor-services gem (DefaultObjectRights class) and adapted to work here, with the `'none'` check added.

File: app/services/cocina/to_fedora/access.rb [hunk 5]
```diff
@@ -23,12 +23,42 @@ def apply
...
+      end
+
+      def license_code
+        license_codes.fetch(access.license)
+      end
+
+      def license_codes
+        DefaultRights::LICENSE_CODES.merge(
+          Cocina::FromFedora::Access::NONE_LICENSE_URI => 'none'
+        )
```
Comment (mjgiarlo) [inline app/services/cocina/to_fedora/access.rb:60]: Here's where we patch in the ability to map nones across.

File: app/services/cocina/from_fedora/access.rb (hunk 1 above)
Comment (jcoyne) [reply app/services/cocina/from_fedora/access.rb:7]: @andrewjbtw I find this confusing. Is this saying "no creativecommons license selected", "in copyright" or "undetermined"? Can we get more specific?

File: app/services/cocina/from_fedora/access.rb [hunk 6]
```diff
@@ -20,7 +22,9 @@ def props
           access: access_rights,
           download: download? ? access_rights : 'none',
           readLocation: location,
-          license: license_uri
+          license: license_uri,
+          copyright: copyright,
```
Comment (jcoyne) [inline app/services/cocina/from_fedora/access.rb:26]: @andrewjbtw please confirm that we want to allow collections to have copyright statements.  I'm not sure collections qualify as creative works that can be copyrighted.

Comment (mjgiarlo) [reply app/services/cocina/from_fedora/access.rb:26]: @jcoyne I believe he confirmed this elsewhere.

File: app/services/cocina/to_fedora/access.rb [hunk 7]
```diff
@@ -23,12 +23,42 @@ def apply
...
+        lookup_and_assign_license! if access.license
         item.rightsMetadata.ng_xml_will_change!
       end
 
       private
 
       attr_reader :item, :access
+
+      def lookup_and_assign_license! # rubocop:disable Metrics/AbcSize
+        if Dor::CreativeCommonsLicenseService.key?(license_code)
```
Comment (jcoyne) [inline app/services/cocina/to_fedora/access.rb:37]: why bother with `license_code` in here and not lookup by the URI in`access.license`.  If we do this, then we can add the CC 4.0 licenses as something that can be round tripped.

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/services/cocina/to_fedora/access.rb (hunk 7 above)
Comment (mjgiarlo) [reply app/services/cocina/to_fedora/access.rb:37]: @jcoyne Currently we do look up the URI in `access.license` in `DefaultRights::LICENSE_CODES`, and then we use the license code to grab the label (and the URI but we can prolly drop that), and we also drop the license code into the Fedora XML.

I think the code as currently written on the ToFedora side has symmetry with the FromFedora side (both use the underlying Dor license services and associated lookup hashes), so I'm currently meh on making this change in this PR, anyway.

File: app/services/cocina/from_fedora/access.rb (hunk 6 above)
Comment (jcoyne) [reply app/services/cocina/from_fedora/access.rb:26]: Can you track that down?

Comment (mjgiarlo) [reply app/services/cocina/from_fedora/access.rb:26]: @jcoyne no problem: https://github.com/sul-dlss/dor-services-app/issues/2623#issuecomment-811203233

**Pull request merged**